tests/unit/models/primo.js

```diff
@@ -0,0 +1,680 @@
+    });
+  });
+
+  describe("primo model getResult method >", function() {
+    it("should include a result object from prmSearchResultAvailabilityLineAfterController $scope", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0096-6762", "0028-4793"]
+            }
+          }
+        }
+      };
+
+      var result = primo.getResult(scope);
+      expect(result).toBeDefined();
+    });
+
+    it("should include a result object from prmSearchResultThumbnailContainerAfterController $scope", function() {
+      var scope = {
+        item: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0096-6762", "0028-4793"]
+            }
+          }
+        }
+      };
+
+      var result = primo.getResult(scope);
+      expect(result).toBeDefined();
+    });
+  });
+
+  describe("primo model urlRewrite method >", function() {
+    it("should rewrite the public api domain", function() {
+      var url = "https://api.thirdiron.com/public/v1/libraries/XXX";
+      expect(primo.urlRewrite(url)).toEqual("https://public-api.thirdiron.com/public/v1/libraries/XXX");
+    });
+
+    it("should not rewrite the public api domain when the public-api domain already exists", function() {
+      var url = "https://public-api.thirdiron.com/public/v1/libraries/XXX";
+      expect(primo.urlRewrite(url)).toEqual("https://public-api.thirdiron.com/public/v1/libraries/XXX");
+    });
+  });
+
+  describe("primo model shouldEnhance method >", function() {
+    it("should not enhance a search result without scope data", function() {
+      var scope = {};
+
+      expect(primo.shouldEnhance(scope)).toEqual(false);
+    });
+
+    it("should enhance a journal search result with an issn", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0096-6762", "0028-4793"]
+            }
+          }
+        }
+      };
+
+      expect(primo.shouldEnhance(scope)).toEqual(true);
+    });
+
+    it("should enhance a journal search result with an eissn, but no issn", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              eissn: ["0096-6762"]
+            }
+          }
+        }
+      };
+
+      expect(primo.shouldEnhance(scope)).toEqual(true);
+    });
+
+    it("should enhance a journal search result with both an issn and an eissn", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0028-4793"],
+              eissn: ["0096-6762"]
```

Comment:
What if we add a test case where one of these values matches our data, but one does not? (simulating the instance where they have an ISSN or eISSN that we simply don’t have)

We should still enhance it, just want to make sure it enhances it when only one of the many provided match.

Reply:
Happy to say we already do! That pretty much describes the journal result acceptance test in a nutshell. Specifically, see lines:
https://github.com/thirdiron/browzine-summon-adapter/pull/28/files#diff-2b01e2be49b9c65a2479fe3472439f6fR67 

https://github.com/thirdiron/browzine-summon-adapter/pull/28/files#diff-2b01e2be49b9c65a2479fe3472439f6fR61

https://github.com/thirdiron/browzine-summon-adapter/pull/28/files#diff-2b01e2be49b9c65a2479fe3472439f6fR75

Notice what you describe is pretty much an end-to-end flow and touches multiple methods, perfect for an acceptance test.

Great question!